libretro-common/include/libretro.h

```diff
@@ -976,6 +976,19 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+struct retro_resource
+{
+   const char *rel_path;
+   const char *base_path;
```

Comment:
I'm not sure if I like base_path being returned as a const char * here. If the path is not static it has to be allocated somewhere, and if the resource directory can depend on the input data is seems like we need to pass down a char * + size_t pair in this struct instead.